# U.S. Medical Insurance Costs

***

###### This is an unguided project from Codecademy as a part of the Data Scientist Career Path. 

### First, we must sort the data from the .CSV file format into a python dictionary.

In [93]:
import csv
insurance_list = [] # Empty list to parse the data into

with open('insurance.csv', newline='') as insurance_csv:
    fields = ['age', 'sex', 'bmi', 'children', 'smoker', 'region', 'charges']
    insurance_reader = csv.DictReader(insurance_csv, delimiter=',')

    patient_age = []
    patient_sex = []
    patient_bmi = []
    patient_children = []
    patient_smoker = []
    patient_region = []
    patient_charges = []

    for row in insurance_reader:
        patient_age.append(row['age'])
        patient_sex.append(row['sex'])
        patient_bmi.append(row['bmi'])
        patient_children.append(row['children'])
        patient_smoker.append(row['smoker'])
        patient_region.append(row['region'])
        patient_charges.append(row['charges'])

***

### Now that the data is parsed and sorted we can start the analyses.

##### What is the average age of the policy holders?

In [94]:
def find_average_age():
    total_age = 0
    for i in patient_age:
        total_age += float(i)
    float_average_age = total_age / len(patient_age)
    average_age = round(float_average_age, 1)
    print(average_age)
find_average_age()


39.2


##### It is determined that the average age of policy holders is 39.2 years.
***


##### Where do the majority of policy holders live?

In [95]:
def region_function(patient_region):
    southwest = 0
    northwest = 0
    southeast = 0
    northeast = 0 
    for i in patient_region:
        if i == 'southwest':
            southwest += 1
        elif i == 'northwest':
            northwest += 1
        elif i == 'southeast':
            southeast += 1
        elif i == 'northeast':
            northeast += 1
    print(str(southwest) + ' policy holders live in the southwest.')
    print(str(northwest) + ' policy holders live in the northwest.')
    print(str(southeast) + ' policy holders live in the southeast.')
    print(str(northeast) + ' policy holders live in the northeast.')
region_function(patient_region)


325 policy holders live in the southwest.
325 policy holders live in the northwest.
364 policy holders live in the southeast.
324 policy holders live in the northeast.


##### It is determined by the function above that 325 policy holders live in the North and Southwest, 324 policy holders live in northeast, and the slight majority live in the southeast of the United States.

***

##### What is the difference in cost between policy holders who smoke and those who do not smoke?

In [96]:
smokers_cost = []
non_smokers_cost = []

for i in range(0, len(patient_age)):
    if patient_smoker[i] == 'yes':
        smokers_cost.append(patient_charges[i])
    elif patient_smoker[i] == 'no':
        non_smokers_cost.append(patient_charges[i])

# Determine total cost for smokers and find the average
total_smokers_cost = 0
for cost in smokers_cost:
    total_smokers_cost += float(cost)

flt_avg_smokers_cost = total_smokers_cost / len(smokers_cost)
avg_smokers_cost = round(flt_avg_smokers_cost, 2)

# Determine total cost for non-smokers and find the average.
total_non_smokers_cost = 0
for cost in non_smokers_cost:
    total_non_smokers_cost += float(cost)

flt_avg_non_smokers_cost = total_non_smokers_cost / len(non_smokers_cost)
avg_non_smokers_cost = round(flt_avg_non_smokers_cost, 2)

avg_diff_smoker_cost = avg_smokers_cost - avg_non_smokers_cost

print('The average cost for policy holders who smoke is $' + str(avg_smokers_cost))
print('The average cost for policy holders who do no smoke is $' + str(avg_non_smokers_cost))
print('Average increase: ' + str(avg_diff_smoker_cost))

The average cost for policy holders who smoke is $32050.23
The average cost for policy holders who do no smoke is $8434.27
Average increase: 23615.96


##### The data shows that on average policy holders who smoke pay $23615.96 more than policy holders who do not smoke.

***

##### What is the average BMI difference between males and females of the policy holders?

In [97]:
# Determine total males and females for average
total_males = []
total_females = []

for i in patient_sex:
    if i == 'male':
        total_males.append(i)
    elif i == 'female':
        total_females.append(i)

# Determine total BMI for each sex
total_male_bmi = 0
total_female_bmi = 0

for i in range(0, len(patient_age)):
    if patient_sex[i] == 'male':
        total_male_bmi += float(patient_bmi[i])
    elif patient_sex[i] == 'female':
        total_female_bmi += float(patient_bmi[i])

# Determine averages and round
flt_avg_male_bmi = total_male_bmi / len(total_males)
flt_avg_female_bmi = total_female_bmi / len(total_females)
avg_male_bmi = round(flt_avg_male_bmi, 1)
avg_female_bmi = round(flt_avg_female_bmi, 1)

print('The average male BMI is ' + str(avg_male_bmi) + '.')
print('The average female BMI is ' + str(avg_female_bmi) + '.')

The average male BMI is 30.9.
The average female BMI is 30.4.


##### The data provided shows little to no difference in BMI when comparing males to females.

***

# Conclusion

- #### The average age of policy holders is 39.2.
- #### The policy holders for the most part live all over the country with a slight majority living in the southeast of the United States.
- #### Policy holders who smoke pay on average $23,615.96 more than policy holders who do not smoke.
- #### The average BMI showed little difference when comparing the two opposite sexes.